Source/WebCore/css/StyleProperties.cpp

```diff
@@ -1669,8 +1664,6 @@ static constexpr bool canUseShorthandForLonghand(CSSPropertyID shorthandID, CSSP
 
     // FIXME: If font-variant-ligatures is none, this depends on the value of the longhand.
     case CSSPropertyFontVariant:
-    // FIXME: Serialization ignores marker-mid and marker-end (http://webkit.org/b/248308).
-    case CSSPropertyMarker:
```

Comment:
If we return an empty string for the shorthand, have you checked that all the longhands appear? (I assume yes)

Reply:
Yes, `cssText` doesn't use a shorthand whose serialization is a null string, which is the case when `getCommonValue()` doesn't find a common value.

```cpp
            if (!value.isNull()) {
                propertyID = shorthandPropertyID;
                shorthandPropertyUsed.set(shorthandPropertyIndex);
                break;
            }
```

```js
var {style} = document.createElement("div");
style.cssText = 'marker-start: url("a"); marker-mid: url("a"); marker-end: url("a")';
style.marker; // 'url("a")'
style.cssText; // 'marker: url("a");'
style.markerEnd = 'url("b")';
style.marker; // ''
style.cssText; // 'marker-start: url("a"); marker-mid: url("a"); marker-end: url("b");'
```